src/frontend/components/DashboardVideoLivePairing/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { Meter, Text } from 'grommet';
+import React, { useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { DashboardButton } from 'components/DashboardPaneButtons/DashboardButtons';
+import { usePairingVideo } from 'data/queries';
+import { Video } from 'types/tracks';
+
+const messages = defineMessages({
+  pairingSecretLabel: {
+    defaultMessage: 'Pair an external device',
+    description:
+      'Button label to request a secret for pairing an external device.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretLabel',
+  },
+  pairingSecretDisplay: {
+    defaultMessage: 'Pairing secret: {secret}',
+    description:
+      'Message displaying the secret for pairing an external device.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretDisplay',
+  },
+  pairingSecretCountdownExpired: {
+    defaultMessage: 'Pairing secret expired',
+    description:
+      'Message displayed when the secret for pairing an external device has expired.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretCountdownExpired',
+  },
+});
+
+interface DashboardVideoLivePairingProps {
+  video: Video;
+}
+
+export const DashboardVideoLivePairing = ({
+  video,
+}: DashboardVideoLivePairingProps) => {
+  const [buttonColor, setButtonColor] = useState('brand');
+  const [secret, setSecret] = useState('');
+  const [expiresIn, setExpiresIn] = useState(0);
+  const [secondsLeft, setSecondsLeft] = useState(0);
+  const [percentsLeft, setPercentsLeft] = useState(100);
+  const [expired, setExpired] = useState(false);
+
+  const usePairingVideoMutation = usePairingVideo(video.id, {
+    onSuccess: (data) => {
+      setSecret(data.secret);
+      setExpiresIn(data.expires_in);
+      setSecondsLeft(data.expires_in);
+      setPercentsLeft(100);
+      setExpired(false);
+      setButtonColor('status-ok');
+    },
+  });
+
+  const pairVideoAction = () => {
+    usePairingVideoMutation.mutate();
+  };
+
+  const upddateStates = () => {
+    setSecondsLeft(secondsLeft - 1);
+    setPercentsLeft((secondsLeft * 100) / expiresIn);
+
+    switch (secondsLeft) {
+      case 0:
+        setExpired(true);
+        setButtonColor('status-warning');
+        break;
+      case -3:
```

Comment:
Is it possible to add comment explaining -3 and -4 to make things clearer to understand please